report/metrics.go

```diff
@@ -52,28 +53,55 @@ type Sample struct {
 	Value     float64   `json:"value"`
 }
 
-var nilMetric = Metric{Samples: ps.NewList()}
+// MakeSingletonMetric makes a metric with a single value
+func MakeSingletonMetric(t time.Time, v float64) Metric {
```

Comment:
Ah. hmm. Could possibly do `func MakeMetric(samples ...Sample) Metric`, which would let you: `MakeMetric()`, `MakeMetric(samples...)`, and `MakeMetric(Sample{t, v})`. Not sure how variadic functions in go perform, though?